### DISCUSSION TITLE
Genshin/HSR/ZZZ 圣遗物/遗器交换格式

### DISCUSSION THREAD

**Discussion by LaoshuBaby** [UIGF-org/.github#10] [Ideas]
最近HSR终于进入到后期开始刷圣遗物的阶段（以下就不挨个改口了统称圣遗物，大家知道是什么就好），然后开始眼花缭乱，想起之前原神在yas里面扫描的时候（但是看起来yas没有任何关于星穹铁道的issue？），搜了很久也没有发现星穹铁道有类似的扫描器

之前也看有 https://github.com/orgs/UIGF-org/discussions/3 背包物品格式 的提案（很好也没有明确是在哪个游戏因此应该是针对Genshin+HSR？），而且有很多背包物品可能是限时的，开发者不一定见过（而且是否可阅读这些变化因素很大）

而圣遗物/遗器这个领域，确实有玩家练度<!-- 其实不喜欢这个词 --规划和后期培养需要（摩拉可能没那么缺，但信用点一般是很缺的，HSR的圣遗物养成成本很高，后续很多玩家开始刷圣遗物以后或许也会有配装难题？）

尤其是目前的调研结果来看，原神这边的不同圣遗物分析已经有多种格式了

https://ideless.github.io/artifact/tutorial/ （[mona](https://github.com/wormtql/genshin_artifact)/genmo/good），导出器主要有[yas](https://github.com/wormtql/yas)/yas-lock/[天目](https://github.com/daydreaming666/Amenoma)）

~~以上，全都是在胡咧咧！一点也不正经！其他大佬尽管修改本段即可！~~

---
2024.01.24 编辑

还有 刻晴办公桌 https://github.com/SkeathyTomas/genshin_artifact_auxiliary

**Comment by BTMuli** [comment]
原神圣遗物**详细数据**获取，目前看来就ocr跟enka两种，

后者暂且不提，数据结构已经稳定了，

前者依赖模型训练，时间成本高而且更新慢，天目已经快1年没更新了，

所以能用的也就Yas跟Enka

因此如果要提出一个应用间圣遗物数据迁移的标准的话，必定得支持三种数据导入，

相比 UIAF 基本靠 YaeAchievement，UIGF 基本靠 api 获取 gachalog，这个新标准，姑且称之为 UIBF(bushi)，麻烦很多

~~不如考虑全变成enka格式~~

**Reply by Lightczx** [reply]
圣遗物理应并入UIIF

**Reply by LaoshuBaby** [reply]
![图片](https://github.com/UIGF-org/.github/assets/42690037/e1ac1673-2b12-4cd5-93fa-ed231ac422a9)
enka是说的zhe玩意吗？enka这个根本没导出而且只有展示栏里面三个角色身上挂的吧，和yas那种ocr全覆盖一遍差远了

~~虽然UIIF那个物品格式感觉现在八字没一撇，但是最统一的方案或许就是冗余量最大的（~~

**Reply by BTMuli** [reply]
> enka这个根本没导出

可以跑api获取数据，[API Doc](https://github.com/EnkaNetwork/API-docs/blob/master/api_chs.md)
